### PR TITLE
The Science Hub app for research is now available for science employees in addition to heads of staff.

### PR DESCRIPTION
**Pull request by zxaber**
## About The Pull Request
- Refractored how access is handled for NTOS apps to be a list. These are handled such that if any required access entry matches access of the user's ID, access is granted.
- The Science Hub app now uses two access entries; Head of Staff or Research access. This allows Science to access the app as well as heads of staff, without making it fully open-access.

## Why It's Good For The Game
Now that the research console is back in science, there isn't any real reason to restrict this app from being usable by science employees. 

## Changelog

:cl:
balance: The Science Hub app can now be downloaded and launched by science employees.
/:cl:

I also re-ordered the `can_run()` proc on programs to check for the guaranteed early `return TRUE` conditions *before* doing any other work.

### CODE REVIEW

**Comment by cacogen** [comment]
Remove it from heads of staff that aren’t the RD while you’re at it

File: code/modules/modular_computers/file_system/programs/techweb.dm [hunk 1]
```diff
@@ -7,8 +7,8 @@
 	size = 16
 	tgui_id = "NtosTechweb"
 	program_icon = "atom"
-	required_access = ACCESS_HEADS
-	transfer_access = ACCESS_RD
+	required_access = list(ACCESS_HEADS, ACCESS_RESEARCH)
+	transfer_access = list(ACCESS_RESEARCH)
```
Comment (JohnFulpWillard) [inline code/modules/modular_computers/file_system/programs/techweb.dm:11]: should keep this as RD-only and make scientists spawn with tablets that have the app. RD is still expected to supervise research so giving them a way to remove certain people from researching would be nice.

**Comment by JohnFulpWillard** [comment]
> Remove it from heads of staff that aren’t the RD while you’re at it

image

**Comment by zxaber** [comment]
- Unless told to by a maintainer, I'd like to keep it available to all heads of staff.

- The eventual plan is for all non-heads to lose roundstart tablets (in favor of a Ntos-style PDA) and so I don't want to add features that would inhibit that. Scientists can download it manually, it doesn't take that long.

**Comment by cacogen** [comment]
while it wasn't a serious suggestion for this pr, it's never made sense that random heads of staff can swoop in and research whatever they want

**Comment by zxaber** [comment]
I think it's fine, it lets other departments get their tech if science is busy/unresponsive/unhelpful.

Research logs are available at the servers (and I guess an app to show them in a tablet wouldn't be terrible) so the RD can always get names to go after if need be.

File: code/modules/modular_computers/file_system/programs/techweb.dm (hunk 1 above)
Comment (UnokiAs) [inline code/modules/modular_computers/file_system/programs/techweb.dm:11]: Changing ACCESS_RESEARCH to ACCESS_RND would avoid paramedic having access to it.

Comment (zxaber) [reply code/modules/modular_computers/file_system/programs/techweb.dm:11]: Works for me

**Pull request merged**